2-Browsers/Week1/homework/ex5-catWalk/index.js

```diff
@@ -21,8 +21,46 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/2-Brow
 
    https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif
 -----------------------------------------------------------------------------*/
+const walkingCat = document.querySelector('img');
+let left = 0;
+const width = 300;
+walkingCat.style.left = left + 'px';
+walkingCat.style.width = width + 'px';
+
 function catWalk() {
-  // TODO complete this function
+  //call an inner  function with setInterval
+  let walk = setInterval(walkBreakPoints, 50);
+  // make another function to set all the walking breakpoints
+  function walkBreakPoints() {
+    const windowWidth = parseInt(window.innerWidth);
+    const halfOfTheScreen = parseInt(windowWidth / 2);
+    //stop at the middle for 5 seconds
+    if (
+      left + width >= halfOfTheScreen &&
+      left + width <= halfOfTheScreen + 10
+    ) {
+      walkingCat.src =
+        'https://media1.tenor.com/images/2de63e950fb254920054f9bd081e8157/tenor.gif';
+      clearInterval(walk);
+
+      window.setTimeout(() => {
+        left += 10;
+        walkingCat.src =
+          'http://www.anniemation.com/clip_art/images/cat-walk.gif';
+        walk = setInterval(walkBreakPoints, 50);
+      }, 5000);
+      //reset the left at the end
+    } else if (left >= windowWidth - 10 && left <= windowWidth) {
+      left = 0;
+
+      //normal situation
+    } else {
+      left += 10;
```

Comment:
```suggestion
    } else if (left >= windowWidth - 10 && left <= windowWidth) {
      //reset the left at the end
      left = 0;
    } else {
      //normal situation
      left += 10;
```

Make sure the comments are in the correct block of code!